manifests/params.pp

```diff
@@ -58,4 +60,13 @@
       fail("Module ${module_name} is not supported on osfamily '${::osfamily}'")
     }
   }
+	case $::architecture {
+    'x86_64', 'amd64': { $arch = 'amd64' }
+    'i386':            { $arch = '386'   }
+    /^arm.*/:          { $arch = 'arm'   }
+    default:           {
+      fail("Unsupported kernel architecture: ${::architecture}")
```

Comment:
In the scenario that I made a custom build of vault for an architecture that doesn't exist here and specified it via `download_url` the module wouldn't allow me to install it.

i.e. this test case would fail:

```
let(:facts) {{
  :architecture => 'ppc'
}}
let(:params) {{
  :download_url => 'https://example.com/vault-ppc.zip'
}}
```

Admittedly it's a bit of an edge case but is introduced by this change.